src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,16 @@
+package duke;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```

Comment:
maybe you should explore the use of constants here in order to reduce the need of magic strings.